mods/page_revisions.js

```diff
@@ -335,14 +448,33 @@ PRS.prototype.fetchAndStoreMWRevision = function(restbase, req) {
 PRS.prototype.getTitleRevision = function(restbase, req) {
     var self = this;
     var rp = req.params;
+    var title = rbUtil.normalizeTitle(rp.title);
     var revisionRequest;
+    var pageDataRequest = restbase.get({
+        uri: self.pageTableURI(rp.domain),
+        body: {
+            table: self.pageTableName,
+            attributes: {
+                title: title
+            }
```

Comment:
For page undeletes we need to iterate back in history and find the previous delete, so for that part we need the full history. However indeed we could make it fetch full history only for `no-cache` requests when page undulation is checked.